cartridges/openshift-origin-cartridge-jbossews/bin/control

```diff
@@ -87,13 +87,26 @@ function ishttpup() {
     return 1
 }
 
-# This function is here to mimmic the EAP cartriges design: 
-# Due to the lack of a provided mgmt interface, the options to check for 
-#  deployment activation is limited. Thus this fucntion uses an ENV VAR to 
-#  allow you configure a sleep time (in seconds), giving your deployments 
-#  time to start before marking the gear as active. 
+# This function is here to mimmic the EAP cartriges design:
+# Due to the lack of a provided mgmt interface, the options to check for
+#  deployment activation is limited. Thus this fucntion uses an ENV VAR to
+#  allow you configure a sleep time (in seconds), giving your deployments
+#  time to start before marking the gear as active.
 function waitondeployments() {
-    sleep ${OPENSHIFT_JBOSSEWS_START_DELAY:-0}
+    attempts=${OPENSHIFT_JBOSSEWS_START_TRYS:-5}
+    sleep_time=${OPENSHIFT_JBOSSEWS_HEALTH_WAIT_TIME:-2}
+    context=${OPENSHIFT_HAPROXY_HTTPCHK_URI:-/}          # context must start with '/'
```

Comment:
`OPENSHIFT_HAPROXY_HTTPCHK_URI` will be present exactly when the framework cartridge is collocated with a proxy, right? And the framework cartridge will be collocated with the proxy on exactly the first gear in non-HA applications, and on some but generally not all subsequent gears with an HA application.